packages/blade/src/components/TopNav/_decisions/decisions.md

```diff
@@ -0,0 +1,291 @@
+
+> The TabNav will also support responsiveness by horizontally scrolling when the items can no longer fit in the available space.
+
+```ts
+type TabNav = {
+  children: React.ReactNode;
+};
+
+type TabNavItem = {
+  /**
+   * href of the link
+   */
+  href?: LinkProps['href'];
+  /**
+   * Anchor tag `target` attribute
+   */
+  target?: LinkProps['target'];
+  /**
+   * as prop to pass ReactRouter's Link component.
+   *
+   * @default 'a'
+   *
+   * @example
+   * ```jsx
+   * import { Link } from 'react-router-dom';
+   *
+   * <TabNavItem as={Link} />
+   * ```
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  as: React.ComponentType<any>;
+  /**
+   * Selected state of the navigation item.
+   *
+   * @default false
+   */
+  isActive?: boolean;
+  /**
+   * Element to render before the navigation item.
+   *
+   * @default undefined
+   */
+  leading?: IconComponent;
+  /**
+   * Element to render inside the navigation item.
+   *
+   * This can either be a string or JSX element (eg: Menu component)
+   */
+  children?: React.ReactNode;
+  /**
+   * Accessibility label for the navigation item.
+   */
+  accessibilityLabel?: string;
+};
+```
+
+TabNav can also contain a `Menu` component which will render a menu when hovered.
+
+```jsx
+const WithMenu = () => {
+  const [isOpen, setOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState<string | null>(null);
+
+  return (
+    <TabNav>
+      <TabNavItem as={RouterLink} isSelected href="/home" icon={HomeIcon} />
+      <Menu interactionType="hover">
+        <TabNavItem 
+          trailingIcon={isOpen ? ChevronDown : ChevronUp}
+        >
+          {selected ? `Others: ${selected}` : "Others"}
+        </TabNavItem>
+        <MenuOverlay>
+          <MenuItem onClick={() => setSelected("Engage")}>Engage</MenuItem>
+          <MenuItem onClick={() => setSelected("Payroll")}>Payroll</MenuItem>
+        </MenuOverlay>
+      </Menu>
+    </TabNav>
+  );
+}
+```
+
+### TabNav overflow behaviour
+
+Once the available space for TabNavItems are less, the TabNav will start to horizontally scroll and will show a chevron icon to indicate that there are more items to the left or right.
+
+https://github.com/razorpay/blade/assets/35374649/8a0fb6ae-cedd-4065-80ed-27647938453c
+
+
+## Mobile UX
+
+On mobile, the TabNav component will no longer be present and the TopNavContent will contain a single button to toggle the side navigation. The TopNavActions will contain the search input, buttons, and user profile.
+
+**Q:** So where will the product navigation items go? 
+**Ans:** The product navigation items will be moved to the bottom of the screen, but this will be part of a separate component called `BottomNavigation` which we will ship in phase 2.
+
+Mobile Navbar example:
+
+```jsx
+const Dashboard = () => {
+  const { isMobile } = useBreakpoint();
+
+  if (isMobile) {
```

Comment:
suggestion: can we toggle the layout based on css media queries instead? I understand this is common pattern and offers some benefits(especially for CSR only apps), however this may not work well for SSR applications. 

Given majority of our apps uses CSR, this may not be a concern for now, just calling it out.

Reply:
This is a contrived example for demonstration purposes, so I've just used isMobile prop. 


And this will be handle on consumer side, they can choose to render the layout however they like. via isMobile or via the `Box` responsive props. And whatever will be controlled from blade internally (eg: size of left nav) will use responsive media props only.

Reply:
Will probably update this example just in case other people copy paste it from here and use it as a pattern.